client/gitSensor/GitSensorGrpcClient.go

```diff
@@ -0,0 +1,185 @@
+package gitSensor
+
+import (
+	"context"
+	"fmt"
+	"github.com/caarlos0/env"
+	pb "github.com/devtron-labs/protos/gitSensor"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"time"
+)
+
+const (
+	ContextTimeoutInSeconds = 10
+	MaxMsgSizeBytes         = 20 * 1024 * 1024
+)
+
+type GitSensorGrpcClient interface {
+	GetChangesInRelease(ctx context.Context, req *pb.ReleaseChangeRequest) (*GitChanges, error)
+}
+
+type GitSensorGrpcClientImpl struct {
+	logger        *zap.SugaredLogger
+	config        *GitSensorGrpcClientConfig
+	serviceClient pb.GitSensorServiceClient
+}
+
+func NewGitSensorGrpcClientImpl(logger *zap.SugaredLogger, config *GitSensorGrpcClientConfig) *GitSensorGrpcClientImpl {
+	return &GitSensorGrpcClientImpl{
+		logger: logger,
+		config: config,
+	}
+}
+
+// getGitSensorServiceClient initializes and returns gRPC GitSensorService client
+func (client *GitSensorGrpcClientImpl) getGitSensorServiceClient() (pb.GitSensorServiceClient, error) {
+	if client.serviceClient == nil {
+		conn, err := client.getConnection()
+		if err != nil {
+			return nil, err
+		}
+		client.serviceClient = pb.NewGitSensorServiceClient(conn)
+	}
+	return client.serviceClient, nil
+}
+
+// getConnection initializes and returns a grpc client connection
+func (client *GitSensorGrpcClientImpl) getConnection() (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), ContextTimeoutInSeconds*time.Second)
+	defer cancel()
+
+	// Configure gRPC dial options
+	var opts []grpc.DialOption
+	opts = append(opts,
+		grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(MaxMsgSizeBytes),
```

Comment:
grpc server restricts incoming payload size to 4Mb. since we are not changing it on the server side we should not set different value at client side.